core/services/keeper/orm.go

```diff
@@ -132,6 +161,17 @@ ORDER BY upkeep_registrations.id ASC, upkeep_registrations.upkeep_id ASC
 	return upkeeps, err
 }
 
+// binaryOfFirstHashInPreviousTurn gets the previous turns starting block hash and converts it to binary
+// we get the previous turns starting block to avoid any contention that could happen over current turns blocks
+func binaryOfFirstHashInPreviousTurn(blockNumber int64, registry Registry, head *types.Head) string {
+	firstHeadInTurn := blockNumber - (blockNumber % int64(registry.BlockCountPerTurn)) - int64(registry.BlockCountPerTurn)
```

Comment:
I'm wondering if it's better to subtract finality depth parameter if we have that? Subtracting BlockCountPerTurn also works, but that would make it future proof if we choose to reduce BlockCountPerTurn to a small number.

Reply:
fair point. changed